neo/Consensus/ConsensusContext.cs

```diff
@@ -174,6 +190,31 @@ private ConsensusPayload MakeSignedPayload(ConsensusMessage message)
             return payload;
         }
 
+        public ConsensusPayload RegenerateSignedPayload(ConsensusMessage message, ushort validatorIndex,
+            byte[] witnessInvocationScript)
+        {
+            message.ViewNumber = ViewNumber;
+            ConsensusPayload payload = new ConsensusPayload
+            {
+                Version = Version,
+                PrevHash = PrevHash,
+                BlockIndex = BlockIndex,
+                ValidatorIndex = validatorIndex,
+                Timestamp = Timestamp,
```

Comment:
Bunch of fixes coming in an update shortly. Another issue was this value is only the value for regenerating the `PrepareRequest`, it needs to be different for the `PrepareResponse`. Will be covered in the next update.

Reply:
This issue is fixed now.